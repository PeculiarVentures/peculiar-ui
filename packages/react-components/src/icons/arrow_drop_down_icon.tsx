import React from 'react';

type TIconProps = React.HTMLAttributes<HTMLOrSVGElement>;

export const ArrowDropDownIcon: React.FC<TIconProps> = (props) => (
  <svg
    {...props}
    xmlns="http://www.w3.org/2000/svg"
    fill="none"
    viewBox="0 0 24 24"
    width="24px"
    height="24px"
  >
    <path
      d="M12.39 14.512a.5.5 0 01-.78 0l-2.96-3.7A.5.5 0 019.04 10h5.92a.5.5 0 01.39.812l-2.96 3.7z"
      fill="currentColor"
    />
  </svg>
);
