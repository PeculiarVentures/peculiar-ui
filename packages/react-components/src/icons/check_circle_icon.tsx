import React from 'react';

type IconProps = React.HTMLAttributes<HTMLOrSVGElement>;

export const CheckCircleIcon: React.FC<IconProps> = (props) => (
  <svg
    {...props}
    xmlns="http://www.w3.org/2000/svg"
    fill="none"
    viewBox="0 0 24 24"
    width="24px"
    height="24px"
  >
    <rect
      x="1.333"
      y="1.333"
      width="21.333"
      height="21.333"
      rx="10.667"
      fill="currentColor"
    />
    <path
      d="M8 10.667L11.333 14l5-5"
      stroke="#fff"
      strokeWidth="1.5"
    />
  </svg>
);
