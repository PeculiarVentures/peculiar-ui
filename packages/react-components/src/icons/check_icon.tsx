import React from 'react';

type TIconProps = React.HTMLAttributes<HTMLOrSVGElement>;

export const CheckIcon: React.FC<TIconProps> = (props) => (
  <svg
    {...props}
    xmlns="http://www.w3.org/2000/svg"
    fill="none"
    viewBox="0 0 24 24"
    width="24px"
    height="24px"
  >
    <path
      d="M4.338 11.656l5.98 5.667a1 1 0 001.452-.081l7.144-8.453a.51.51 0 00-.043-.716l-.38-.338a.508.508 0 00-.717.043l-6.84 7.7-5.584-4.96a.509.509 0 00-.717.042l-.337.38a.51.51 0 00.042.716z"
      fill="currentColor"
    />
  </svg>
);
