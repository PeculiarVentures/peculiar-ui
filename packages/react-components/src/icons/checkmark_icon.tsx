import React from 'react';

type IconProps = React.HTMLAttributes<HTMLOrSVGElement>;

export const CheckmarkIcon: React.FC<IconProps> = (props) => (
  <svg
    {...props}
    xmlns="http://www.w3.org/2000/svg"
    fill="none"
    viewBox="0 0 20 20"
    width="20px"
  >
    <path
      fill="currentColor"
      d="M3.6 9.7l4.9 4.6c.4.4 1 .4 1.4 0l5.9-7v-.6l-.4-.3a.4.4 0 00-.6 0L9.1 13 4.5 8.8a.4.4 0 00-.6 0l-.3.3c-.2.2-.2.5 0 .6z"
    />
  </svg>
);
