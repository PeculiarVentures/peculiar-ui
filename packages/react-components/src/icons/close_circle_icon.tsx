import React from 'react';

type IconProps = React.HTMLAttributes<HTMLOrSVGElement>;

export const CloseCircleIcon: React.FC<IconProps> = (props) => (
  <svg
    {...props}
    xmlns="http://www.w3.org/2000/svg"
    fill="none"
    viewBox="0 0 24 24"
    width="24px"
    height="24px"
  >
    <rect
      x="1"
      y="1"
      width="21.333"
      height="21.333"
      rx="10.667"
      fill="currentColor"
    />
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M12.185 11.5l2.721 2.72a.324.324 0 010 .457l-.23.229a.323.323 0 01-.456 0l-2.72-2.72-2.72 2.72a.324.324 0 01-.457 0l-.229-.23a.324.324 0 010-.456l2.721-2.72-2.721-2.72a.324.324 0 010-.457l.229-.229a.324.324 0 01.457 0l2.72 2.72 2.72-2.72a.323.323 0 01.457 0l.229.23a.324.324 0 010 .456l-2.721 2.72z"
      fill="#fff"
    />
  </svg>
);
