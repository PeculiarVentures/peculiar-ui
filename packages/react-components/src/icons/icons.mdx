import { Meta, IconGallery, IconItem } from '@storybook/blocks';
import {
  ArrowDropDownIcon,
  ArrowRightIcon,
  CheckCircleIcon,
  CheckIcon,
  CloseCircleIcon,
  CloseIcon,
  CloseSmallIcon,
  DotIcon,
  PlusIcon,
  WarningIcon,
} from './index';

<Meta title="Components/Icons" />

# Icons

<IconGallery>
  <IconItem name="ArrowDropDown">
    <ArrowDropDownIcon />
  </IconItem>
  <IconItem name="ArrowRight">
    <ArrowRightIcon />
  </IconItem>
  <IconItem name="CheckCircle">
    <CheckCircleIcon />
  </IconItem>
  <IconItem name="Check">
    <CheckIcon />
  </IconItem>
  <IconItem name="CloseCircle">
    <CloseCircleIcon />
  </IconItem>
  <IconItem name="Close">
    <CloseIcon />
  </IconItem>
  <IconItem name="CloseSmall">
    <CloseSmallIcon />
  </IconItem>
  <IconItem name="Dot">
    <DotIcon />
  </IconItem>
  <IconItem name="Plus">
    <PlusIcon />
  </IconItem>
  <IconItem name="Warning">
    <WarningIcon />
  </IconItem>
</IconGallery>
