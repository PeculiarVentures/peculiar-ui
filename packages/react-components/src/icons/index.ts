export { CheckIcon } from './check_icon';
export { InfoIcon } from './info_icon';
export { CloseIcon } from './close_icon';
export { CloseSmallIcon } from './close_small_icon';
export { ArrowDropDownIcon } from './arrow_drop_down_icon';
