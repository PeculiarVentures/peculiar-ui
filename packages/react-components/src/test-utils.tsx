import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
