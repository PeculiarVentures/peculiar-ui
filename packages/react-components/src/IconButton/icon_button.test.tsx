import React from 'react';
import { renderWithWrapper as render, screen } from '../test-utils';
import { IconButton } from './index';
import { PlusIcon } from '../icons';

describe('<IconButton />', () => {
  it('should render as default', () => {
    const { asFragment } = render(
      <IconButton>
        <PlusIcon role="img" />
      </IconButton>,
    );

    const button = screen.getByRole('button');
    const icon = screen.getByRole('img');

    expect(button).toBeInTheDocument();
    expect(icon).toBeInTheDocument();
    expect(button.tagName).toBe('BUTTON');
    expect(icon.tagName).toBe('svg');
    expect(button).toContainElement(icon);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should have title', () => {
    const { asFragment } = render(
      <IconButton title="Test title">
        <PlusIcon />
      </IconButton>,
    );

    const button = screen.getByRole('button');

    expect(button.getAttribute('aria-label')).toMatch(/Test title/i);

    expect(asFragment()).toMatchSnapshot();
  });
});
