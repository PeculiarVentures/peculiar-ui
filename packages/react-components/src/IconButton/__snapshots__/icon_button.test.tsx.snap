// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<IconButton /> > colors > color "default" 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  border: 1px solid transparent;
  -webkit-transition: background-color 200ms,color 200ms,box-shadow 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,box-shadow 200ms,border-color 200ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  vertical-align: middle;
  background-color: transparent;
  padding: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: var(--pv-color-black);
  border-radius: 4px;
  height: var(--pv-size-base-7);
  min-width: var(--pv-size-base-7);
  --pv-color-black: var(--pv-color-gray-10);
  border: none;
}

.emotion-0:disabled {
  cursor: not-allowed;
  box-shadow: none;
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled):hover {
  background-color: var(--pv-color-gray-3);
}

.emotion-0:not(:disabled):focus-visible {
  background-color: var(--pv-color-gray-4);
}

.emotion-0:not(:disabled):active {
  background-color: var(--pv-color-gray-5);
}

.emotion-1 {
  margin: 0;
  color: inherit;
  font-weight: var(--pv-text-btn1-weight);
  font-size: var(--pv-text-btn1-size);
  line-height: var(--pv-text-btn1-height);
  letter-spacing: var(--pv-text-btn1-spacing);
  width: 100%;
  display: inherit;
  -webkit-align-items: inherit;
  -webkit-box-align: inherit;
  -ms-flex-align: inherit;
  align-items: inherit;
  -webkit-box-pack: inherit;
  -ms-flex-pack: inherit;
  -webkit-justify-content: inherit;
  justify-content: inherit;
}

<button
    class="emotion-0"
    type="button"
  >
    <span
      class="emotion-1"
    >
      <svg
        fill="none"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 12.5h12M12 18.5v-12"
          stroke="currentColor"
          stroke-width="1.5"
        />
      </svg>
    </span>
  </button>
</DocumentFragment>
`;

exports[`<IconButton /> > colors > color "primary" 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  border: 1px solid transparent;
  -webkit-transition: background-color 200ms,color 200ms,box-shadow 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,box-shadow 200ms,border-color 200ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  vertical-align: middle;
  background-color: transparent;
  padding: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: var(--pv-color-primary);
  border-radius: 4px;
  height: var(--pv-size-base-7);
  min-width: var(--pv-size-base-7);
  --pv-color-black: var(--pv-color-gray-10);
  border: none;
}

.emotion-0:disabled {
  cursor: not-allowed;
  box-shadow: none;
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled):hover {
  background-color: var(--pv-color-primary-tint-5);
}

.emotion-0:not(:disabled):focus-visible {
  background-color: var(--pv-color-primary-tint-4);
}

.emotion-0:not(:disabled):active {
  background-color: var(--pv-color-primary-tint-3);
}

.emotion-1 {
  margin: 0;
  color: inherit;
  font-weight: var(--pv-text-btn1-weight);
  font-size: var(--pv-text-btn1-size);
  line-height: var(--pv-text-btn1-height);
  letter-spacing: var(--pv-text-btn1-spacing);
  width: 100%;
  display: inherit;
  -webkit-align-items: inherit;
  -webkit-box-align: inherit;
  -ms-flex-align: inherit;
  align-items: inherit;
  -webkit-box-pack: inherit;
  -ms-flex-pack: inherit;
  -webkit-justify-content: inherit;
  justify-content: inherit;
}

<button
    class="emotion-0"
    type="button"
  >
    <span
      class="emotion-1"
    >
      <svg
        fill="none"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 12.5h12M12 18.5v-12"
          stroke="currentColor"
          stroke-width="1.5"
        />
      </svg>
    </span>
  </button>
</DocumentFragment>
`;

exports[`<IconButton /> > colors > color "secondary" 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  border: 1px solid transparent;
  -webkit-transition: background-color 200ms,color 200ms,box-shadow 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,box-shadow 200ms,border-color 200ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  vertical-align: middle;
  background-color: transparent;
  padding: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: var(--pv-color-secondary);
  border-radius: 4px;
  height: var(--pv-size-base-7);
  min-width: var(--pv-size-base-7);
  --pv-color-black: var(--pv-color-gray-10);
  border: none;
}

.emotion-0:disabled {
  cursor: not-allowed;
  box-shadow: none;
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled):hover {
  background-color: var(--pv-color-secondary-tint-5);
}

.emotion-0:not(:disabled):focus-visible {
  background-color: var(--pv-color-secondary-tint-4);
}

.emotion-0:not(:disabled):active {
  background-color: var(--pv-color-secondary-tint-3);
}

.emotion-1 {
  margin: 0;
  color: inherit;
  font-weight: var(--pv-text-btn1-weight);
  font-size: var(--pv-text-btn1-size);
  line-height: var(--pv-text-btn1-height);
  letter-spacing: var(--pv-text-btn1-spacing);
  width: 100%;
  display: inherit;
  -webkit-align-items: inherit;
  -webkit-box-align: inherit;
  -ms-flex-align: inherit;
  align-items: inherit;
  -webkit-box-pack: inherit;
  -ms-flex-pack: inherit;
  -webkit-justify-content: inherit;
  justify-content: inherit;
}

<button
    class="emotion-0"
    type="button"
  >
    <span
      class="emotion-1"
    >
      <svg
        fill="none"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 12.5h12M12 18.5v-12"
          stroke="currentColor"
          stroke-width="1.5"
        />
      </svg>
    </span>
  </button>
</DocumentFragment>
`;

exports[`<IconButton /> > colors > color "white" 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  border: 1px solid transparent;
  -webkit-transition: background-color 200ms,color 200ms,box-shadow 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,box-shadow 200ms,border-color 200ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  vertical-align: middle;
  background-color: transparent;
  padding: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: var(--pv-color-white);
  border-radius: 4px;
  height: var(--pv-size-base-7);
  min-width: var(--pv-size-base-7);
  --pv-color-black: var(--pv-color-gray-10);
  border: none;
}

.emotion-0:disabled {
  cursor: not-allowed;
  box-shadow: none;
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled):hover {
  background-color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled):focus-visible {
  background-color: var(--pv-color-gray-8);
}

.emotion-0:not(:disabled):active {
  background-color: var(--pv-color-gray-9);
}

.emotion-1 {
  margin: 0;
  color: inherit;
  font-weight: var(--pv-text-btn1-weight);
  font-size: var(--pv-text-btn1-size);
  line-height: var(--pv-text-btn1-height);
  letter-spacing: var(--pv-text-btn1-spacing);
  width: 100%;
  display: inherit;
  -webkit-align-items: inherit;
  -webkit-box-align: inherit;
  -ms-flex-align: inherit;
  align-items: inherit;
  -webkit-box-pack: inherit;
  -ms-flex-pack: inherit;
  -webkit-justify-content: inherit;
  justify-content: inherit;
}

<button
    class="emotion-0"
    type="button"
  >
    <span
      class="emotion-1"
    >
      <svg
        fill="none"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 12.5h12M12 18.5v-12"
          stroke="currentColor"
          stroke-width="1.5"
        />
      </svg>
    </span>
  </button>
</DocumentFragment>
`;

exports[`<IconButton /> > colors > color "wrong" 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  border: 1px solid transparent;
  -webkit-transition: background-color 200ms,color 200ms,box-shadow 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,box-shadow 200ms,border-color 200ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  vertical-align: middle;
  background-color: transparent;
  padding: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: var(--pv-color-wrong);
  border-radius: 4px;
  height: var(--pv-size-base-7);
  min-width: var(--pv-size-base-7);
  --pv-color-black: var(--pv-color-gray-10);
  border: none;
}

.emotion-0:disabled {
  cursor: not-allowed;
  box-shadow: none;
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled):hover {
  background-color: var(--pv-color-wrong-tint-5);
}

.emotion-0:not(:disabled):focus-visible {
  background-color: var(--pv-color-wrong-tint-4);
}

.emotion-0:not(:disabled):active {
  background-color: var(--pv-color-wrong-tint-3);
}

.emotion-1 {
  margin: 0;
  color: inherit;
  font-weight: var(--pv-text-btn1-weight);
  font-size: var(--pv-text-btn1-size);
  line-height: var(--pv-text-btn1-height);
  letter-spacing: var(--pv-text-btn1-spacing);
  width: 100%;
  display: inherit;
  -webkit-align-items: inherit;
  -webkit-box-align: inherit;
  -ms-flex-align: inherit;
  align-items: inherit;
  -webkit-box-pack: inherit;
  -ms-flex-pack: inherit;
  -webkit-justify-content: inherit;
  justify-content: inherit;
}

<button
    class="emotion-0"
    type="button"
  >
    <span
      class="emotion-1"
    >
      <svg
        fill="none"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 12.5h12M12 18.5v-12"
          stroke="currentColor"
          stroke-width="1.5"
        />
      </svg>
    </span>
  </button>
</DocumentFragment>
`;

exports[`<IconButton /> > should have title 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  border: 1px solid transparent;
  -webkit-transition: background-color 200ms,color 200ms,box-shadow 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,box-shadow 200ms,border-color 200ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  vertical-align: middle;
  background-color: transparent;
  padding: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: var(--pv-color-black);
  border-radius: 4px;
  height: var(--pv-size-base-7);
  min-width: var(--pv-size-base-7);
  --pv-color-black: var(--pv-color-gray-10);
  border: none;
}

.emotion-0:disabled {
  cursor: not-allowed;
  box-shadow: none;
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled):hover {
  background-color: var(--pv-color-gray-3);
}

.emotion-0:not(:disabled):focus-visible {
  background-color: var(--pv-color-gray-4);
}

.emotion-0:not(:disabled):active {
  background-color: var(--pv-color-gray-5);
}

.emotion-1 {
  margin: 0;
  color: inherit;
  font-weight: var(--pv-text-btn1-weight);
  font-size: var(--pv-text-btn1-size);
  line-height: var(--pv-text-btn1-height);
  letter-spacing: var(--pv-text-btn1-spacing);
  width: 100%;
  display: inherit;
  -webkit-align-items: inherit;
  -webkit-box-align: inherit;
  -ms-flex-align: inherit;
  align-items: inherit;
  -webkit-box-pack: inherit;
  -ms-flex-pack: inherit;
  -webkit-justify-content: inherit;
  justify-content: inherit;
}

<button
    aria-label="Test title"
    class="emotion-0"
    type="button"
  >
    <span
      class="emotion-1"
    >
      <svg
        fill="none"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 12.5h12M12 18.5v-12"
          stroke="currentColor"
          stroke-width="1.5"
        />
      </svg>
    </span>
  </button>
</DocumentFragment>
`;

exports[`<IconButton /> > should pass className 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  border: 1px solid transparent;
  -webkit-transition: background-color 200ms,color 200ms,box-shadow 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,box-shadow 200ms,border-color 200ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  vertical-align: middle;
  background-color: transparent;
  padding: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: var(--pv-color-black);
  border-radius: 4px;
  height: var(--pv-size-base-7);
  min-width: var(--pv-size-base-7);
  --pv-color-black: var(--pv-color-gray-10);
  border: none;
}

.emotion-0:disabled {
  cursor: not-allowed;
  box-shadow: none;
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled):hover {
  background-color: var(--pv-color-gray-3);
}

.emotion-0:not(:disabled):focus-visible {
  background-color: var(--pv-color-gray-4);
}

.emotion-0:not(:disabled):active {
  background-color: var(--pv-color-gray-5);
}

.emotion-1 {
  margin: 0;
  color: inherit;
  font-weight: var(--pv-text-btn1-weight);
  font-size: var(--pv-text-btn1-size);
  line-height: var(--pv-text-btn1-height);
  letter-spacing: var(--pv-text-btn1-spacing);
  width: 100%;
  display: inherit;
  -webkit-align-items: inherit;
  -webkit-box-align: inherit;
  -ms-flex-align: inherit;
  align-items: inherit;
  -webkit-box-pack: inherit;
  -ms-flex-pack: inherit;
  -webkit-justify-content: inherit;
  justify-content: inherit;
}

<button
    class="my-class-name emotion-0"
    type="button"
  >
    <span
      class="emotion-1"
    >
      <svg
        fill="none"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 12.5h12M12 18.5v-12"
          stroke="currentColor"
          stroke-width="1.5"
        />
      </svg>
    </span>
  </button>
</DocumentFragment>
`;

exports[`<IconButton /> > should render as default 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  border: 1px solid transparent;
  -webkit-transition: background-color 200ms,color 200ms,box-shadow 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,box-shadow 200ms,border-color 200ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  vertical-align: middle;
  background-color: transparent;
  padding: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: var(--pv-color-black);
  border-radius: 4px;
  height: var(--pv-size-base-7);
  min-width: var(--pv-size-base-7);
  --pv-color-black: var(--pv-color-gray-10);
  border: none;
}

.emotion-0:disabled {
  cursor: not-allowed;
  box-shadow: none;
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled):hover {
  background-color: var(--pv-color-gray-3);
}

.emotion-0:not(:disabled):focus-visible {
  background-color: var(--pv-color-gray-4);
}

.emotion-0:not(:disabled):active {
  background-color: var(--pv-color-gray-5);
}

.emotion-1 {
  margin: 0;
  color: inherit;
  font-weight: var(--pv-text-btn1-weight);
  font-size: var(--pv-text-btn1-size);
  line-height: var(--pv-text-btn1-height);
  letter-spacing: var(--pv-text-btn1-spacing);
  width: 100%;
  display: inherit;
  -webkit-align-items: inherit;
  -webkit-box-align: inherit;
  -ms-flex-align: inherit;
  align-items: inherit;
  -webkit-box-pack: inherit;
  -ms-flex-pack: inherit;
  -webkit-justify-content: inherit;
  justify-content: inherit;
}

<button
    class="emotion-0"
    type="button"
  >
    <span
      class="emotion-1"
    >
      <svg
        fill="none"
        height="24px"
        role="img"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 12.5h12M12 18.5v-12"
          stroke="currentColor"
          stroke-width="1.5"
        />
      </svg>
    </span>
  </button>
</DocumentFragment>
`;
